### PR TITLE
payload-testing-prow-plugin: add some debugging logs

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/test-platform

### CODE REVIEW

File: cmd/payload-testing-prow-plugin/server.go [hunk 1]
```diff
@@ -176,11 +184,16 @@ func (s *server) handle(l *logrus.Entry, ic github.IssueCommentEvent) string {
 		builder.spec = spec
 		var jobNames []string
 		var jobTuples []api.MetadataWithTest
+		specLogger.Debug("resolving jobs ...")
+		startResolveJobs := time.Now()
 		jobs, err := s.jobResolver.resolve(spec.ocp, spec.releaseType, spec.jobs)
+		specLogger.WithField("duration", time.Since(startResolveJobs)).Debug("resolving jobs completed")
```
Comment (petr-muller) [inline cmd/payload-testing-prow-plugin/server.go:190]: Could we log how many jobs do we have received?

File: cmd/payload-testing-prow-plugin/server.go [hunk 2]
```diff
@@ -199,16 +212,23 @@ func (s *server) handle(l *logrus.Entry, ic github.IssueCommentEvent) string {
 				jobTuples = append(jobTuples, jobTuple)
 			}
 		}
+		specLogger.WithField("duration", time.Since(startResolveTests)).Debug("resolving tests completed")
```
Comment (petr-muller) [inline cmd/payload-testing-prow-plugin/server.go:215]: Could we log how many jobs did we manage to resolve?

File: cmd/payload-testing-prow-plugin/server.go [hunk 3]
```diff
@@ -199,16 +212,23 @@ func (s *server) handle(l *logrus.Entry, ic github.IssueCommentEvent) string {
...
 				specLogger.WithError(err).Error("could not create PullRequestPayloadQualificationRun")
 				return fmt.Sprintf("could not create PullRequestPayloadQualificationRun: %v", err)
 			}
+			specLogger.WithField("duration", time.Since(startCreateRuns)).WithField("runNames", runNames).Debug("creating PullRequestPayloadQualificationRuns completed")
 		} else {
 			specLogger.Warn("found no resolved tests")
 		}
 		messages = append(messages, message(spec, jobNames))
 	}
+	logger.WithField("duration", time.Since(start)).Debug("handle completed")
```
Comment (petr-muller) [inline cmd/payload-testing-prow-plugin/server.go:231]: Would it make sense to publish all these durations as metrics?

Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go:232]: I think yes if we have more profiling to come.
In that case, we just need to switch those duration to a histogram and expose it to prow-monitoring.

File: cmd/payload-testing-prow-plugin/server.go (hunk 1 above)
Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go]: Done.

File: cmd/payload-testing-prow-plugin/server.go (hunk 2 above)
Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go]: Done.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**